### ISSUE TITLE
Do not use same name for index and foreign key constraint.

### ISSUE THREAD

**Issue by Sjlver** [strapi/strapi#19624]
TL;DR: In its database schema, Strapi uses the same name for both a *foreign key constraint* and an *index* related to that constraint. This breaks other tools (like Prisma) which assume that names are unique for a given table.

### Required System information

- Node.js version: 20.5.1
- Strapi version: 4.17.0
- Database: postgres

### Describe the bug

When Strapi creates its database schema, it adds foreign key constraints to tables. It also adds indexes to speed up the evaluation of these foreign key constraints. Strapi uses the same name for both the foreign key and the index. This is allowed by postgres, but breaks other tools which assume that names are unique within each table.

Here is a shortened schema produced by Strapi:

```sql
-- Strapi's admin_users table.
CREATE TABLE public.admin_users (
    id integer NOT NULL,
    firstname character varying(255),
    -- many columns omitted
    created_by_id integer
);

-- The index, called admin_users_created_by_id_fk.
CREATE INDEX admin_users_created_by_id_fk ON public.admin_users USING btree (created_by_id);

-- The foreign key constraint, using the same name.
ALTER TABLE ONLY public.admin_users
    ADD CONSTRAINT admin_users_created_by_id_fk FOREIGN KEY (created_by_id) REFERENCES public.admin_users(id) ON DELETE SET NULL;
```

Other tools choke on this. We tried to use access Strapi's database using Prisma. Prisma produces the following error: `Error parsing attribute "@relation": The given constraint name "admin_users_created_by_id_fk" has to be unique in the following namespace: on model "admin_users" for primary key, indexes, unique constraints and foreign keys. Please provide a different name using the "map" argument.`

### Steps to reproduce the behavior

1. Install Strapi and let it initialize the database
2. Initialize prisma: `npx prisma init`
3. Configure prisma's `DATABASE_URL` to point to Strapi's database
4. Run `npx prisma db pull` to introspect the database. This will produce a schema.prisma file similar to the one below, with the duplicate constraint
5. Run `npx prisma generate`. This command fails with the error message given above.

Prisma schema file:

```prisma
model admin_users {
  id                Int           @id @default(autoincrement())
  firstname         String?       @db.VarChar(255)
  created_by_id     Int?
  admin_users       admin_users?  @relation("admin_usersToadmin_users", fields: [created_by_id], references: [id], onUpdate: NoAction, map: "admin_users_created_by_id_fk")

  @@index([created_by_id], map: "admin_users_created_by_id_fk")
}

```

### Expected behavior

Strapi should use a different name for its indexes (a common practice is to use the suffix `_ix`) and its foreign key constraints (which use the `_fk` suffix).

**Comment by HermanBide** [comment]
has this been fixed ?

**Comment by derrickmehaffy** [comment]
Will be addressed in Strapi 5, but slightly unrelated to us, we didn't use Prisma and it does work fine with Knex. I agree it's bad practice but we are changing the way we generate table/column/etc names in v5.

Closing as this won't change in v4.

**Comment by alfoadd** [comment]
This is still an issue in Strapi v5
